src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -539,55 +520,56 @@ private PropertyPatternClauseSyntax ParsePropertyPatternClause()
                 closeToken: out SyntaxToken closeBraceToken,
                 openKind: SyntaxKind.OpenBraceToken,
                 closeKind: SyntaxKind.CloseBraceToken,
-                out bool isPropertyPatternClause);
-            var kind = isPropertyPatternClause ? SyntaxKind.PropertyPatternClause : SyntaxKind.ListPatternClause;
-            return _syntaxFactory.PropertyPatternClause(kind, openBraceToken, subPatterns, closeBraceToken);
+                static p => p.ParseSubpatternElement());
+            return _syntaxFactory.PropertyPatternClause(openBraceToken, subPatterns, closeBraceToken);
         }
 
-        private void ParseSubpatternList(
+        private void ParseSubpatternList<T>(
             out SyntaxToken openToken,
-            out SeparatedSyntaxList<SubpatternSyntax> subPatterns,
+            out SeparatedSyntaxList<T> subPatterns,
             out SyntaxToken closeToken,
             SyntaxKind openKind,
             SyntaxKind closeKind,
-            out bool isPropertyPatternClause)
+            Func<LanguageParser, T> parseFunc)
```

Comment:
>Func<LanguageParser, T> parseFunc [](http://example.com/codeflow?start=12&length=33)

Rather than a func which is one of two possibilities, consider just making this a flag. This doesn't feel like the type of thing that we need to expand in the future, so the func feels needlessly complicated and splitting out functionality to too many locations.

Reply:
This is mainly because it's a generic method (we want to parse either a subpattern or a pattern), I don't think it could be simplified further.

Reply:
I'm not really understanding why this needs to be generic either. Why do we need to parse subpatterns in one case and patterns in another case? I'd expect all cases to be the same.

Reply:
This is the same method used to parse a list of SubpatternSyntax. We want a list of PatternSyntax in the list pattern. So I figured to make it generic.